applications/serial_lte_modem/src/dfu/slm_at_dfu.c

```diff
@@ -200,8 +206,12 @@ static int download_client_callback(const struct download_client_evt *event)
 		}
 
 		if (!first_fragment) {
-			rsp_send("\r\n#XDFUGET: %d,%d\r\n", dfu_step,
-				(file_download_size * 100) / file_image_size);
+			uint32_t percentage = 0;
+
+			percentage = (file_download_size * 100) / file_image_size;
+			if ((percentage % PERCENTAGE_STEP) == 0) {
```

Comment:
I guess the point in here is that we wont write DFUGET for all the fragments if % is not even rising, but is there now a risk that with certain file image size and fragment size, this will very rarely be true? If fragment is small enough, so that this is hit multiple times for 1% of file_image_size, then this is ok, but if fragment is larger than 1%, this is not hit every 5th time. Also, if fragment is smaller than 1%, I think that this will be hit multiple times.

Would it be better to keep track which % we showed last and update if the difference is big enough?

Reply:
This URC was pop-up every fragment. If the cellular speed is slow or image is larger, the same % will show up multiple times. The point here is to reduce redundant progress. You're right. We might miss few progress URC if size is not even.
The progress doesn't need to be accuracy as long as we got 100% progress in the end.

It's okay to keep track % and send URC once difference is big enough. What step do you think is better?

Reply:
I think that it would be better that we would only update the % when there is something to update. So that if current % is larger than previous we would show it. If needed, this could have some threshold, such as, only show when difference is larger than 5%, but show the real %.

However, I do not know whether what end user expects to see, so treat this as a suggestion.